src/simsopt/_core/dofs.py

```diff
@@ -44,39 +43,47 @@ class Dofs:
     that have been combined from multiple optimizable objects, keeping
     only the non-fixed dofs.
 
+    For the meaning of ``abs_step`` and ``rel_step``, see
+    :func:`simsopt._core.util.finite_difference_steps()`.
+
     Args:
+        funcs: A list, tuple, or set of optimizable functions.
         fail: Should be None, a large positive float, or NaN. If not
           None, any ObjectiveFailure excpetions raised will be caught
           and the corresponding residual values will be replaced by this
           value.
-    """
-
-    def __init__(self,
-                 funcs,
-                 fail: Union[None, float] = None):
-        """
-        Given a list of optimizable functions, 
-
-        funcs: A list/set/tuple of callable functions.
-
-        returns: an object with the following attributes:
-        x: A 1D numpy vector of variable dofs.
+        abs_step: Absolute step size for finite differences.
+        rel_step: Relative step size for finite differences.
+        differences: Method to use if and when finite differences are
+          evaluated. Should be ``forward`` or ``centered``.
 
+    Attributes:
         dof_owners: A vector, with each element pointing to the object whose
-        set_dofs() function should be called to update the corresponding
-        dof.
+          set_dofs() function should be called to update the corresponding
+          dof.
 
         all_owners: A list of all objects involved in computing funcs,
-        including those that do not directly own any of the non-fixed
-        dofs.
+          including those that do not directly own any of the non-fixed
+          dofs.
 
         indices: A vector of ints, with each element giving the index in
-        the owner's set_dofs method corresponding to this dof.
+          the owner's set_dofs method corresponding to this dof.
 
         names: A list of strings to identify each of the dofs.
-        """
+    """
+
+    def __init__(self,
+                 funcs,
+                 fail: Union[None, float] = 1.0e12,
+                 abs_step: float = 1.0e-7,
+                 rel_step: float = 0.0,
+                 differences: str = "centered"
```

Comment:
Instead of differences, differences_method or differences_approach or the short names diff_method or diff_approach would be more appropriate.